DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,557 @@
+		footstep = 0
+	else
+		footstep++
+
+/obj/item/clothing/suit/space/borg/equipped(mob/user, slot)
+	. = ..()
+	if (slot == SLOT_WEAR_SUIT)
+		if (mobhook && mobhook.parent != user)
+			QDEL_NULL(mobhook)
+		if (!mobhook)
+			mobhook = user.AddComponent(/datum/component/redirect, list(COMSIG_MOVABLE_MOVED = CALLBACK(src, .proc/on_mob_move)))
+	else
+		QDEL_NULL(mobhook)
+
+/obj/item/clothing/suit/space/borg/dropped()
+	. = ..()
+	QDEL_NULL(mobhook)
+
+/obj/item/clothing/suit/space/borg/Destroy()
+	QDEL_NULL(mobhook) // mobhook is not our component
+	return ..()
+
+/obj/item/clothing/suit/space/borg/Initialize()
+	. = ..()
+	icon_state = "borg[pick(1,2)]"
+
+
+/obj/effect/mob_spawn/human/alive/trek/borg
+	name = "Borg drone"
+	assignedrole = "borg drone"
+	outfit = /datum/outfit/borg
+
+
+/obj/structure/chair/borg
+	name = "If you see this, tell a coder"
+	desc = "Stop looking at me."
+	max_integrity = 100
+
+/obj/structure/chair/borg/conversion
+	name = "assimilation bench"
+	desc = "Submit yourself to the collective"
+	icon_state = "borg_off"
+	icon = 'DS13/icons/borg/borg.dmi'
+	anchored = TRUE
+	can_buckle = TRUE
+	can_be_unanchored = FALSE
+	max_buckled_mobs = TRUE
+	buildstacktype = null
+	item_chair = null // if null it can't be picked up
+	var/restrained = FALSE //can they unbuckle easily?
+
+
+/obj/structure/chair/borg/conversion/user_buckle_mob(mob/living/carbon/human/M, mob/user)
+	if(!isBorgDrone(M))
+		to_chat(M, "Assimilate them first")
+		return
+	M.unequip_everything()
+	icon_state = "borg_off"
+	visible_message("parts of [src] start to shift and move")
+	playsound(loc, 'DS13/sound/effects/borg/machines/convert_table.ogg', 50, 1, -1)
+	var/image/armoverlay = image('DS13/icons/borg/borg.dmi')
+	armoverlay.icon_state = "borg_arms"
+	armoverlay.layer = ABOVE_MOB_LAYER
+	overlays += armoverlay
+	var/image/armoroverlay = image('DS13/icons/borg/borg.dmi')
+	armoroverlay.icon_state = "borgarmour"
+	armoroverlay.layer = ABOVE_MOB_LAYER
+	overlays += armoroverlay
+	sleep(35)
+	playsound(loc, 'DS13/sound/effects/borg/machines/convert_table2.ogg', 50, 1, -1)
+	icon_state = "borg_off"
+	M.equipOutfit(/datum/outfit/borg, visualsOnly = FALSE) //Outfit handles name etc.
+	cut_overlays()
+	overlays -= armoverlay
+	overlays -= armoroverlay
+	qdel(armoroverlay)
+	qdel(armoverlay)
+	return TRUE
+
+
+/obj/structure/chair/borg/charging
+	name = "recharging alcove"
+	desc = "We must recharge to regain"
+	icon_state = "borgcharger"
+	icon = 'DS13/icons/borg/borg.dmi'
+	anchored = TRUE
+	can_buckle = TRUE
+	can_be_unanchored = FALSE
+	max_buckled_mobs = 1
+	resistance_flags = FIRE_PROOF
+	buildstacktype = null
+	item_chair = null // if null it can't be picked up
+	var/cooldown2 = 110 //music loop cooldowns
+	var/saved_time2 = 0
+	var/sound = 'DS13/sound/effects/borg/machines/alcove.ogg'
+	var/mob/living/carbon/human/user
+
+/obj/structure/chair/borg/charging/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
```

Comment:
@spookydonut apparently Kmc cant find a way outside of Init to get this working correctly. Got any ideas?

Reply:
Yeah so unless I use a while loop, the start and stop processing doesn't want to work